pkg/proxy/proxy.go

```diff
@@ -150,11 +151,14 @@ func (proxy *Proxy) cleanup(ctx context.Context) {
 		proxy.CleanSettings.CronSchedule)
 
 	proxy.runGarbageCollection(ctx)
-	for proxy.removeTags() {
+	remove, _ := proxy.shouldRemoveTags()
+	iteration := 0
+
+	for remove {
 		lruImages := proxy.Cache.GetLruList()
 		common.Log.Infof("total tags: %d", len(lruImages))
-		removalTags := int(float64(len(lruImages)) * proxy.CleanSettings.CleanTagsPercentage)
-		common.Log.Infof("Removing %d tags", removalTags)
+		removalTags := int(math.Max(math.Round(float64(len(lruImages))*proxy.CleanSettings.CleanTagsPercentage), math.Min(float64(iteration), 1)))
```

Comment:
minor suggestion - break this into two clearly named variables because the line is really hard to read with the additions, something like:

```
thing1 := math.Round(float64(len(lruImages)) * proxy.CleanSettings.CleanTagsPercentage)
thing2 := math.Min(float64(iteration), 1)
removalTags := int(math.Max(thing1, thing2))
```
but actual meaningful names :D